arm-billing/2017-02-27-preview/swagger/billing.json

```diff
@@ -0,0 +1,399 @@
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+                "$ref": "#/definitions/Invoice"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Billing/operations": {
+      "get": {
+        "tags": [
+          "Operations"
+        ],
+        "operationId": "Operations_List",
+        "description": "Lists all of the available billing REST API operations.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/OperationListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "DownloadUrl": {
+      "description": "A secure URL that can be used to download a PDF invoice until the URL expires.",
+      "properties": {
+        "expiryTime": {
+          "description": "The time in UTC at which this download URL will expire.",
+          "type": "string",
+          "format": "date-time",
+          "readOnly": true
+        },
+        "url": {
+            "description": "The URL to the PDF file.",
+            "type": "string",
+            "readOnly": true
+        }
+      }
+    },
+    "ErrorDetails": {
+      "description": "The details of the error.",
+      "properties": {
+        "code": {
+          "description": "Error code.",
+          "type": "string",
+          "readOnly": true
+        },
+        "message": {
+          "description": "Error message indicating why the operation failed.",
+          "type": "string",
+          "readOnly": true
+        },
+        "target": {
+          "description": "The target of the particular error.",
+          "type": "string",
+          "readOnly": true
+        }
+      }      
+    },
+    "ErrorResponse": {
+      "description": "Error reponse indicates that the service is not able to process the incoming request. The reason is provided in the error message.",
+      "type": "object",
+      "properties": {
+        "error": {
+           "$ref": "#/definitions/InvoiceProperties"
+        }
+      }
+    },
+    "Invoice": {
```

Comment:
The definition currently provided is correct. However, that will have side effects on some of Autorest generated clients (ex: Java, Ruby). When a model is tagged with extension `"x-ms-azure-resource": true`, Java and Ruby generators do not generate that model and use the model from the runtime. 

Hence, I would recommend:
> "Invoice" should be an allOf on the "Resource".

Here is an [example](https://github.com/Azure/azure-rest-api-specs/blob/master/arm-compute/2016-04-30-preview/swagger/compute.json#L5277) the standard Resource definition example as per ARM recommendation.

An example of how "[VirtualMachine" is an allOf on the "Resource"](https://github.com/Azure/azure-rest-api-specs/blob/master/arm-compute/2016-04-30-preview/swagger/compute.json#L4187).